core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java

```diff
@@ -130,11 +130,8 @@ private void sendConfirmationCodeIfNumberUpdated(UserDto user, User userEntity,
 	{
 		Optional<ConfirmationCode> confirmationCode = createConfirmationCodeIfNumberUpdated(user.getMobileNumber(),
 				originalUser.getMobileNumber(), userEntity);
-		if (confirmationCode.isPresent())
-		{
 			sendConfirmationCodeTextMessage(userEntity.getMobileNumber(), confirmationCode.get(),
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "confirmationCode.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS3655)